code/modules/hydroponics/grown.dm

```diff
@@ -475,4 +491,21 @@
 			return
 		reagents.remove_any(rand(1,3)) //Todo, make it actually remove the reagents the seed uses.
 		seed.do_thorns(H,src)
-		seed.do_sting(H,src,pick("r_hand","l_hand"))
\ No newline at end of file
+		seed.do_sting(H,src,pick("r_hand","l_hand"))
+
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/On_Consume()
+	if(seed && seed.get_trait(TRAIT_BATTERY_RECHARGE))
+		if(!reagents.total_volume)
+			var/batteries_recharged = 0
+			for(var/obj/item/weapon/stock_parts/cell/C in usr.GetAllContents())
+				var/newcharge = (potency*0.01)*C.maxcharge
+				if(C.charge < newcharge)
```

Comment:
So rather than add to existing charge (assuming less than maximum capacity), this flat out fails to do anything if your plant has a low potency or the battery already has a charge above the amount you are attempting to charge to.

Seems a bit bothersome, in my opinion, but a neat addition nonetheless. Does it work on batteries that are installed in things (like eguns), or only batteries that are by themselves?